packages/@aws-cdk/aws-s3/lib/bucket.ts

```diff
@@ -1614,15 +1615,21 @@ export class Bucket extends BucketBase {
     }
     Bucket.validateBucketName(bucketName);
 
+    const _regionInfo = regionInfo.RegionInfo.get(region);
+
     const oldUrlFormatRegions = new Set(regionsBefore(RULE_S3_WEBSITE_REGIONAL_SUBDOMAIN));
 
     const newUrlFormat = attrs.bucketWebsiteNewUrlFormat === undefined
       ? !oldUrlFormatRegions.has(region)
       : attrs.bucketWebsiteNewUrlFormat;
 
-    const websiteDomain = newUrlFormat
-      ? `${bucketName}.s3-website.${region}.${urlSuffix}`
-      : `${bucketName}.s3-website-${region}.${urlSuffix}`;
+    const websiteDomain = attrs.bucketWebsiteNewUrlFormat === undefined
+      ? (_regionInfo.s3StaticWebsiteEndpoint !== undefined
+        ? `${bucketName}.${_regionInfo.s3StaticWebsiteEndpoint}`
+        : (oldUrlFormatRegions.has(region)
+          ? `${bucketName}.s3-website-${region}.${urlSuffix}`
+          : `${bucketName}.s3-website.${region}.${urlSuffix}`))
+      : (attrs.bucketWebsiteNewUrlFormat ? `${bucketName}.s3-website.${region}.${urlSuffix}` : `${bucketName}.s3-website-${region}.${urlSuffix}`);
```

Comment:
How about we refactor this using helper function(s) so it becomes readable again 😹 

Something like
```ts
function bucketWebsiteDomain() {
  if (attrs.bucketWebsiteNewUrlFormat === undefined) {
    rerturn doItTheOldWay();
  }

  return doItTheNewWay();
}
```

Reply:
Yes sure, I already did kind of exactly you suggested 😁 my `refactor` commit is still in the build check

Reply:
Haha, I just missed that commit with my review. Looks better now, but I'm not a fan of the higher order functions here, feels unnecessarily clever. In the CDK we usually prefer helper functions + if statements.

Reply:
I absolutely agree, the reason for the higher order functions were that I don't need to pass so many parameters. But I will change it to helper functions if it's preferred in this project 👌